index.html

```diff
@@ -2209,218 +2270,232 @@ <h3><dfn>Writing or pushing content</dfn></h3>
-              </li>
-              <li>
-                Let |timeout:number| be |options|'s timeout.
-              </li>
-              <li>
-                If the |message:NDEFMessageSource| parameter is not of type defined by
-                the <a>NDEFMessageSource</a> union, reject |p|
-                with a {{TypeError}}, and abort these steps.
-              </li>
-              <li>
-                If the |message| parameter is of
-                <a>NDEFMessageInit</a> type, and |message|'s records
-                [= list/is empty =], reject |p| with
-                a {{TypeError}} and abort these steps.
-              </li>
+                <p class="note">
+                  The UA might abort message push at this point. The reasons
+                  for termination are implementation details. For example, the
+                  user could have has set a preference to allow a given origin
+                  only to read, write, or push data to peers. Also, the
+                  implementation might be unable to support the requested
+                  operation.
+                </p>
+                <li>
+                  Let |output| be the notation for the <a>NDEF message</a>
+                  to be created by UA, as the result of passing
+                  |message| to <a>create Web NFC message</a>.
+                  If this throws an exception, reject |p| with that
+                  exception and abort these steps.
+                </li>
+                <li>
+                  Attempt to <a>abort a pending push operation</a>.
+                  <p class="note">
+                    A push replaces all previously configured push operations.
+                  </p>
+                </li>
+                <li>
+                  Set `this`.[[\PushOptions]] to |options|.
+                </li>
+                <li>
+                  Set `this`.[[\PushMessage]] to |output|.
+                </li>
+                <li>
+                  Set <a>pending push tuple</a> to (`this`, |p|).
+                </li>
+                <li>
+                  Run the <a>start the push timer</a> steps.
+                </li>
+                <li>
+                  Run the <a>add write handler</a> steps.
+                </li>
+                <p class="note">
+                  If <a>NFC is suspended</a>,
+                  continue waiting until |timer| expires (if set), or
+                  promise is aborted by the user, or until an <a>NFC device</a>
+                  comes within communication range.
+                </p>
+              </ol>
+            </li>
+          </li>
+        </ol>
+      </p>
+
+      <p id="steps-push-timer">
+        To <dfn>start the push timer</dfn>, run these steps:
+        <ol class=algorithm>
+          <p>
+            Let |p:Promise| be the <a>pending push tuple</a>'s promise.
+          </p>
+          <li>
+            Let |writer| be the <a>pending push tuple</a>'s writer.
```

Comment:
writer:NFCWriter

Reply:
I don't think we can say "'s writer" here because writer is not a known thing. 'promise' is, thought. Maybe say

Let |writer| be the pending push tuple's NFCWriter instance

Reply:
Well, wasn't writer defined for the pending push tuple? Again, I copied the text elsewhere.
If we change style, it's OK, but then we need to modify in every similar place.

Reply:
Yes, because it was needed to reset the 'async write handlers" which is what you are cleaning up - but this really should be handled like I mentioned in other comments